test/DeathTest.cpp

```diff
@@ -144,7 +147,7 @@ TEST(DeathTest, ThreadSafeTest) {
       Death::Instance().RegisterDeathEvent(&DeathTest::RaceTest, "race");
    };
    std::vector<std::future<void>> waitingPromises;
-   for (int i = 0; i < 10000; i++) {
+   for (int i = 0; i < 10; i++) {
```

Comment:
@craig-cogdill : 2 would actually be enough. We just want to see that the registration and death callback works per thread